ublox/src/ubx_packets/packets.rs

```diff
@@ -2135,6 +2138,69 @@ bitflags! {
 //     msss: u32,
 // }
 
+#[derive(Clone)]
+pub struct RxmSfrbxIter<'a> {
+    data: &'a [u8],
+    offset: usize,
+}
+
+impl<'a> core::iter::Iterator for RxmSfrbxIter<'a> {
+    type Item = u32;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.offset < self.data.len() {
+            let data: [u8; 4] = (&self.data[self.offset..self.offset + 4])
+                .try_into()
+                .expect("data incorrect length slice, this should not occur");
+            self.offset += 4;
+            Some(u32::from_ne_bytes(data))
+        } else {
+            None
+        }
+    }
+}
+
+impl fmt::Debug for RxmSfrbxIter<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut clone = self.clone();
+        let dwrd: Vec<u32> = clone.collect();
```

Comment:
Is this Debug necessary? I feel like most Debug impls would not perform a clone and rather just rely on "{:?}" you get by deriving Debug on the struct. Maybe impl a display like method on the struct instead?

Reply:
You should be able to do `f.debug_list().entries(self.clone().iter()).finish()`. Clone is cheap it's just a slice and usize. This way does avoid allocating a vec tho

Reply:
Hmm I think we might want to end up iterating the iterator as how the output is converted, from `[u8; N]` to `[u32; N/4]`. I was avoiding to use clone but self can't be mutable in signature of `Debug`

Not quite sure as well with their previous implementations of iterators: 
- `NavSatIter` - split into multiple parts `NavSatSvInfoRef` 
- `MonVerExtensionIter` - converting to strings 
which only uses `debug_struct("#class_name")` does not output correctly i.e.
`NavSat(NavSat { itow: 250738000, version: 1, num_svs: 35, reserved: [0, 0], svs: NavSatIter })`
`svs:` becomes just the name.

Reply:
Yeah that's pretty typical for debug messages, they aren't really indeed for anything other than debugging. Iterators normally do not print their contents when you debug print them. If anything it should probably just print out the underlying slice and the offset

Reply:
Are you maybe confusing `std::fmt::Debug` for `serde::Serialize`? The debug output here shouldn't matter, you need to do this for all the messages - https://serde.rs/impl-serialize.html. Similarly if a message has one of these iter fields, it will have to run thru the iterator to serialize it

Reply:
oh yea... had another pr for serialize, forgot this is just debugging